codex/rest/api.nim

```diff
@@ -218,27 +218,32 @@ proc initRestApi*(node: CodexNodeRef, conf: CodexConf): RestRouter =
   router.rawApi(
     MethodPost,
     "/api/codex/v1/upload") do (
+      chunk: Option[uint]
     ) -> RestApiResponse:
-      ## Upload a file in a streamming manner
+      ## Upload a file in a streamming manner,
+      ## split it into blocks of `chunk` bytes,
+      ## and save to the node's BlockStore
       ##
 
       trace "Handling file upload"
-      var bodyReader = request.getBodyReader()
+      let bodyReader = request.getBodyReader()
       if bodyReader.isErr():
-        return RestApiResponse.error(Http500)
+        return RestApiResponse.error(Http500, bodyReader.error)
 
-      # Attempt to handle `Expect` header
-      # some clients (curl), wait 1000ms
-      # before giving up
-      #
+      # Attempt to handle `Expect` header some clients (curl),
+      # wait 1000ms before giving up
       await request.handleExpect()
 
-      let
-        reader = bodyReader.get()
+      let bodyStream = AsyncStreamWrapper.new(reader = AsyncStreamReader(bodyReader.get))
 
       try:
-        without cid =? (
-          await node.store(AsyncStreamWrapper.new(reader = AsyncStreamReader(reader)))), error:
+        let runUpload =
+          if (chunkOpt =? chunk) and (blockSize =? chunkOpt) and (blockSize <= MaxBlockSize):
+            node.store(bodyStream, blockSize = blockSize.int)
+          else:
+            node.store(bodyStream)
+
```

Comment:
Current code just ignores any errors in syntax of `?chunk` REST parameter and goes into "else" part i.e. default chunk size in these cases. We may return parsing error message instead, I don't know whether it makes sense?

Reply:
If you look at the top of the file, there is a `validate` function that is currently empty which `presto` uses to validate all requests. I believe we can use it to validate the syntax of `chunk`.

Reply:
We should add a validator for the `chunk` parsing.

Reply:
We can use this function to check that `chunk` has a reasonable value, say between 512 and 1 GB. Does it seem reasonable to you?

Reply:
Yes, that sounds reasonable 👍 

We could potentially (not now, in the future), add specifiers for the size, like `mb`, `kb`, `gb`, etc... Just a thought...